[RHACS][Docs][4.7 and later versions] ROX-28197: Adding docs for ACS Dev Hub Integration (#89111)
openshift/openshift-docs: modules/viewing-security-information-in-red-hat-developer-hub.adoc

// Module included in the following assemblies:
//

> // *configuration/configuring-and-integrating-the-rhacs-plugin-with-red-hat-developer-hub.adoc

Reviewer: ```suggestion
// * configuration/configuring-and-integrating-the-rhacs-plugin-with-red-hat-developer-hub.adoc
```

:_mod-docs-content-type: PROCEDURE
[id="viewing-security-information-in-red-hat-developer-hub_{context}"]
= Viewing security information in Red Hat Developer Hub

By configuring and integrating the {rh-rhacs-first} plugin with {rh-rhdh-first}, you can access vulnerability data, assess risks, and take proactive security actions without leaving the {rh-rhdh} environment.

.Prerequisites

> * You have enabled the {product-title-short} plugin installation in {rh-rhdh}.

Author: it is ok to use present perfect as per the guidelines:
https://redhat-documentation.github.io/supplementary-style-guide/#prerequisites:~:text=You%20have%20validated%20Thing%201.

> * You have enabled the {product-title-short} plugin installation in {rh-rhdh}.

Reviewer: If you feel the present perfect adds something there or makes it clearer, sure, NBD.

+
For more information, see link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{rhdh-latest-version}/html/installing_and_viewing_plugins_in_red_hat_developer_hub/rhdh-installing-rhdh-plugins_title-plugins-rhdh-about#con-install-dynamic-plugin-helm_rhdh-installing-rhdh-plugins[Installing dynamic plugins using the Helm chart].

.Procedure

> . Create an `app-config.yaml` file with the `proxy` and `acs` stanza by using the following content:

Reviewer: ```suggestion
. Create an `app-config.yaml` file that contains `proxy` and `acs` stanzas by using the following content:
```

+
[source,yaml]
----
# ...
proxy:
  endpoints:
    /acs:
      credentials: dangerously-allow-unauthenticated
      target: ${ACS_API_URL}
      headers:
        authorization: Bearer ${ACS_API_KEY}
acs:
  acsUrl: ${ACS_API_URL}
# ...
----

> . To enable the {product-title-short} plugin, perform the following steps:

Reviewer: ```suggestion
. Enable the {product-title-short} plugin:
```

Reviewer: It's a lead-in sentence in either case.

Reviewer: Oh, wait, my mistake. It's not a lead-in sentence to a procedure--it's just a step. https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#writing-steps

Both the original and the suggestion remain acceptable to ISG, though. It's just word golf. 🤷

> .. Navigate to the dynamic plugins configuration file in your {rh-rhdh} setup.

Reviewer: Does this need to be multiple nested steps? Could revise to something like:

```
. Open the dynamic plugins configuration file in your {rh-rhdh} setup and add the following section to it:" 
```

.. To include the {product-title-short} plugin, add the following content to the configuration file, for example:
+

> [source,yaml]

Reviewer: Are there any specifics or replaceable values in this YAML to call out?

Author: nothing that I know of

----
# ...

> - package: https://github.com/RedHatInsights/backstage-plugin-advanced-cluster-security/releases/download/v0.1.1/redhatinsights-backstage-plugin-acs-dynamic-0.1.1.tgz

Reviewer 2: Good point. Would it be possible to mention both until it gets accepted? We can remove the Github artifacts once it's accepted.

>   integrity: sha256-9JeRK2jN/Jgenf9kHwuvTvwTuVpqrRYsTGL6cpYAzn4=

Reviewer 3: I believe this will be removed as well once we switch to using npm packages.

Author: As discussed in chat, we will create a separate ticket to update the doc after the project is accepted in the upstream repository.

  disabled: false
  pluginConfig:
    dynamicPlugins:
      frontend:
        redhatinsights.backstage-plugin-acs:
          entityTabs:
            - path: /acs
              title: RHACS
              mountPoint: entity.page.acs
          mountPoints:
            - mountPoint: entity.page.acs/cards
              importName: EntityACSContent
              config:
                layout:
                  gridColumnEnd:
                    lg: span 12
                    md: span 12
                    xs: span 12
# ...
----

> . To add annotations for entities, perform the following steps:

Reviewer: This is not my area, so I'm probably ignorant here, but will the reader know what "entities" are in this context?

Reviewer 2: We should assume they might not. Some clarification would be good. What do you think?

Author: @sachaudh Could you please suggest a text that would make the context more clear?

Author: https://github.com/RedHatInsights/backstage-plugin-advanced-cluster-security?tab=readme-ov-file#:~:text=Each%20entity%20in%20the%20catalog%20will%20need%20to%20have%20an%20annotation%20added%20that%20references%20the%20deployment(s)%20in%20order%20to%20display%20vulnerability%20data%20for%20them.%20Here%20is%20an%20example%3A

Author: I tried to add more context to entites.

+
[NOTE]
====
To display the vulnerability data, each entity in the RHDH catalog must reference the RHACS deployments.
====

> .. Navigate to the entity configuration file for your service.

Reviewer: Same note here as before. Does navigating to a file need to be its own step?

Reviewer 2: It might be fine to leave it as a separate step. Is there a general form for the file name? Maybe we could include an example.

Author: @maknop Please help us here. I could not find anything more in the readme file.
https://github.com/RedHatInsights/backstage-plugin-advanced-cluster-security?tab=readme-ov-file#:~:text=Each%20entity%20in%20the%20catalog%20will%20need%20to%20have%20an%20annotation%20added%20that%20references%20the%20deployment(s)%20in%20order%20to%20display%20vulnerability%20data%20for%20them.%20Here%20is%20an%20example%3A

[id="viewing-security-information-in-red-hat-developer-hub_{context}"]
= Viewing security information in Red Hat Developer Hub

By configuring and integrating the {rh-rhacs-first} plugin with {rh-rhdh-first}, you can access vulnerability data, assess risks, and take proactive security actions without leaving the {rh-rhdh} environment.

.Prerequisites

* You have enabled the {product-title-short} plugin installation in {rh-rhdh}.
+
For more information, see link:https://docs.redhat.com/en/documentation/red_hat_developer_hub/{rhdh-latest-version}/html/installing_and_viewing_plugins_in_red_hat_developer_hub/rhdh-installing-rhdh-plugins_title-plugins-rhdh-about#con-install-dynamic-plugin-helm_rhdh-installing-rhdh-plugins[Installing dynamic plugins using the Helm chart] ({rh-rhdh} documentation).

.Procedure

. Create an `app-config.yaml` file that contains the `proxy` and `acs` stanzas by using the following content:
+
[source,yaml]
----
# ...
proxy:
  endpoints:
    /acs:
      target: ${ACS_API_URL}
      headers:
        authorization: Bearer ${ACS_API_KEY}
acs:
  acsUrl: ${ACS_API_URL}
# ...
----

. To enable the {product-title-short} plugin, perform the following steps:

> .. Navigate to the dynamic plugins configuration file in your {rh-rhdh} setup, for example, `app-config.yaml`.

Reviewer 3: Sorry, just noticed this! The dynamic plugins configurations file will be different than the `app-config.yaml`. The dynamic plugins config is apart of RHDH and is an additional step for configuring the plugin with RHDH. The app-config.yaml is used for both Backstage and RHDH.

I would just remove this from portion of this sentence: ", for example, `app-config.yaml`."

.. To include the {product-title-short} plugin, add the following content to the configuration file, for example:
+
[source,yaml]
----
# ...
- package: https://github.com/RedHatInsights/backstage-plugin-advanced-cluster-security/releases/download/v0.1.1/redhatinsights-backstage-plugin-acs-dynamic-0.1.1.tgz
  integrity: sha256-9JeRK2jN/Jgenf9kHwuvTvwTuVpqrRYsTGL6cpYAzn4=
  disabled: false
  pluginConfig:
    dynamicPlugins:
      frontend:
        redhatinsights.backstage-plugin-acs:
          entityTabs:
            - path: /acs
              title: RHACS
              mountPoint: entity.page.acs
          mountPoints:
            - mountPoint: entity.page.acs/cards
              importName: EntityACSContent
              config:
                layout:
                  gridColumnEnd:
                    lg: span 12
                    md: span 12
                    xs: span 12
# ...
----

. To add annotations for entities in the {rh-rhdh} catalog, perform the following steps:
+
[NOTE]
====

> To display the vulnerability data, each entity in the {rh-rhdh} catalog must reference the {product-title-short} deployments.

Reviewer 3: I would say the following for this:
`To display the vulnerability data, each component entity in the {rh-rhdh} catalog must reference the {product-title-short} deployments.`

The following values are associated with the entities in the {rh-rhdh} catalog:

** `API`
** `Component`
** `Domain`
** `Group`
** `Location`
** `Resource`
** `System`
** `Template`
** `User`
====
.. Navigate to the entity configuration file for your service in your {rh-rhdh} setup.
.. Add the following annotation to the configuration file, for example:
+
[source,yaml]
----
# ...

> acs/deployment-name: "test-deployment-1,test-deployment-2,test-deployment-3"

Reviewer 3: Instead of just listing the annotation as we have here, lets do this:
```
apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: test-service
  annotations:
      acs/deployment-name: "test-deployment-1,test-deployment-2,test-deployment-3"
...
```
